share/spack/setup-env.sh

```diff
@@ -40,6 +40,16 @@
 ########################################################################
 
 spack() {
+    # Store LD_LIBRARY_PATH variables from spack shell function
+    # This is necessary because MacOS System Integrity Protection clears
+    # (DY?)LD_LIBRARY_PATH variables on process start.
+    if [ -n "${LD_LIBRARY_PATH-}" ]; then
```

Comment:
Shouldn't this be `:-`, not just `-`?

```bash
if [ -n "${LD_LIBRARY_PATH:-}" ]; then
...
```

Reply:
I think in this context they're identical

LD_LIBRARY_PATH set non-zero-length: both evaluate to LD_LIBRARY_PATH
LD_LIBRARY_PATH not set: both evaluate to replacement string, which is the empty string
LD_LIBRARY_PATH set zero length string: Yours evaluates to the replacement string, which is the empty string. Mine evaluates to LD_LIBRARY_PATH, which is the empty string.

Reply:
ok sounds good -- I had to read up on this AGAIN since shell substitution is like *the least* intuitive syntax.